board_view.md

```diff
@@ -0,0 +1,273 @@
+# Board View for WIT Backend Services
+
+Implements User Stories:
+ - Board Configuration on Process Template: https://openshift.io/openshiftio/openshiftio/plan/detail/1725
+ - Basic Board View for Planner: https://openshift.io/openshiftio/openshiftio/plan/detail/2073
+
+## Description
+This implements the backend support for board views. It contains changes to the JSONAPI services as well as to the space template system. It provides backend support for a flexible, extensible board view implementation.
+
+## Changes to the JSONAPI
+
+This section describes the API of the new service changes. It defines new entities available thru JSONAPI as well as the changes to existing endpoints.
+
+### Board/Column Definition in the Space Template Response
+
+The response for `/spacetemplates` and `/spacetemplates/:id` gets a new relationship for the board definition:
+
+```yaml
+# [...]
+"workitemboards": {
+  "links": {
+    "related": "http://api/spacetemplates/000-000-001/workitemboards"
+  }
+},
+# [...]
+```
+
+Note that this relationship will *not* be present on the response for `/space/:id` as having normalized template information on the `space` relationships is deprecated. The client should always use the provided `space-template` relationship on the `space` entitiy to pull template information.
+
+The new external relationship `workitemboards` is available at the endpoint `/spacetemplates/:id/workitemboards` and has the following structure: 
+
+```yaml
+{
+  "id": "000-000-002",
+  "attributes": {
+    "name": "Scenarios Board",
+    "description": "This is the default board config for the legacy template (Experiences).",
+    "contextType": "TypeLevelContext",  // this designates the type of the context
+    "context": "000-000-003",  // this designates the ID of the context, in this case the typegroup ID
+    "created-at": "0001-01-01T00:00:00Z",
+    "updated-at": "0001-01-01T00:00:00Z"
+  },
+  "relationships": {
+    "spaceTemplate": {
+      "data": {
+        "id": "000-000-004",
+        "type": "spacetemplates"
+      }
+    },
+    "columns": {
+      "data": [
+        {
+          "id": "000-000-005",
+          "title": "workitemboardcolumn",
+          "order": 0,  // the left-to-right order of the column in the view
```

Comment:
@michaelkleinhenz will `0` by definition always be the left-most column and will `len(columns)-1` always be the right-most column?